### ISSUE TITLE
ask about invert_canonicalization

### ISSUE THREAD

**Issue by the-full** [arnab39/equiadapt#20]
Hello,
Firstly, I would like to express my admiration for your work on this project.  It has been a great resource for the community.
I have a question regarding the implementation of the "invert_canonicalization" function.  In both the paper and the code, you have provided this function to invert the rotation of the outputs after canonicalization  for the prediction network.  However, in the example of image segmentation, it seems that you have only used the "canonicalize" function by rotating the target ([https://github.com/arnab39/equiadapt/blob/main/examples/images/segmentation/model.py#L121](https://github.com/arnab39/equiadapt/blob/main/examples/images/segmentation/model.py#L121)), without utilizing the "invert_canonicalization" function.  Could you please clarify why the "invert_canonicalization" function is not required in this instance?
Thank you for your time and assistance.  I look forward to your response.

**Comment by sibasmarak** [comment]
Hi @the-full, thank you for your kind and encouraging words!

Ideally, yes, you should use the `invert_canonicalization` function. However, we do not rely on it during the training for two reasons:

1. Our training/experiments for instance segmentation task involves training the canonicalization function only (no fine-tuning of segmentation models) with the prior regularization. Once we train a canonicalization function to learn the mapping to the prior distribution, we can use it with any architecture (in our case, both MaskRCNN and Segment-Anything Model)! Therefore, we do not use the `invert_canonicalization` and set `task_weight` as 0 in our experiments.

2. Additionally, some specific design decisions were taken to avoid using `invert_canonicalization` in the case of (frozen) large pretrained models ([here](https://github.com/arnab39/equiadapt/blob/main/examples/images/segmentation/model.py#L147-L148)). Effectively, if the canonicalizer is trained well, we can expect the output of the prediction network to be identical to the output for the canonical inputs (which is the ground truth because the predominant orientation in the datasets is the canonical one). However, feel free to use the function before computing the loss (and I'd recommend this if you are training from scratch and not using the prior regularization).

During inference, instead of `invert_canonicalization` function, we use custom invert [functions](https://github.com/arnab39/equiadapt/blob/main/equiadapt/images/utils.py) for `masks` and `bboxes`.

If your issues are addressed, please go ahead and close this issue. Thank you!

**Comment by the-full** [comment]
Thank you for your prompt reply, which I believe has answered my question thoroughly.  I will now close this issue and once again, thank you for your time and effort!
